system/pamd.py

```diff
@@ -0,0 +1,418 @@
+- name: Insert a new rule before an existing rule
+  pamd: name=system-auth type=auth control=required module_path=pam_faillock.so new_type=auth new_control=sufficient new_module_path=pam_faillock.so state=before
+- name: Insert a new rule after an existing rule
+  pamd: name=system-auth type=auth control=required module_path=pam_faillock.so new_type=auth new_control=sufficient new_module_path=pam_faillock.so state=after
+- name: Remove module arguments from an existing rule
+  pamd: name=system-auth type=auth control=required module_path=pam_faillock.so module_arguments='' state=updated
+- name: Replace all module arguments in an existing rule
+  pamd: name=system-auth type=auth control=required module_path=pam_faillock.so module_arguments='preauth silent deny=3 unlock_time=604800 fail_interval=900' state=updated
+- name: Remove specific arguments from a rule
+  pamd: name=system-auth type=session control='[success=1 default=ignore]' module_path=pam_succeed_if.so module_arguments='crond quiet' state=args_absent
+- name: Ensure specific arguments are present in a rule
+  pamd: name=system-auth type=session control='[success=1 default=ignore]' module_path=pam_succeed_if.so module_arguments='crond quiet' state=args_present
+- name: Update specific argument value in a rule
+  pamd: name=system-auth type=auth control=required module_path=pam_faillock.so module_arguments='fail_interval=300' state=args_present
+"""
+
+RETURN = '''
+dest:
+    description: path to pam.d service that was changed
+    returned: success
+    type: string
+    sample: "/etc/pam.d/system-auth"
+...
+'''
+
+#The PamdRule class encapsulates a rule in a pam.d service
+class PamdRule(object):
+    def __init__(self, ansible, stringline=None, rule_type=None,
+                 rule_control=None, rule_module_path=None,
+                 rule_module_args=None):
+
+        if stringline is not None and (rule_type is not None or
+                                       rule_control is not None or
+                                       rule_module_path is not None):
+            ansible.fail_json(msg='This is bizarre.  You can only pass' +
+                              'in a line OR a type, control, and module path')
+        elif stringline is not None:
+            self.line = stringline
+            self.__parsline__()
+        else:
+            self.rule_type = rule_type
+            self.rule_control = rule_control
+            self.rule_module_path = rule_module_path
+            
+            try:
+                if self.rule_module_args is not None and type(self.rule_module_args) is list:
+                    self.rule_module_args = rule_module_args
+                elif self.rule_module_args is not None and type(self.rule_module_args) is str:
+                    self.rule_module_args = rule_module_args.split(' ')
+            except AttributeError:
+                self.rule_module_args = []
+
+    def __parsline__(self):
```

Comment:
That is an unusual method name.

Why the leading '__' ? Why the trailing '__'?